src/scipp/interpolate/__init__.py

```diff
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+"""Sub-package for objects used in interpolation."""
+
+from ..core import array, Variable, DataArray, DimensionError, UnitError
+from ..compat.wrapping import wrap1d
+
+from typing import Callable
+
+
+@wrap1d(is_partial=True)
+def interp1d(da: DataArray, dim: str, **kwargs) -> Callable:
```

Comment:
What about the case where the x coord is not a dim-coord? We could allow passing the coord-name instead of a dim name.

Reply:
I think that would make sense, yes, but at the same time I feel that changing the param name to anything but `dim` would be confusing and unexpected. Can we keep it as `dim`, but use it as coord name? Or is that too confusing?

Reply:
We could also use a separate `coord` arg. But then we have mutually exclusive args again...
Positional only?